Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabase.cpp

```diff
@@ -860,7 +860,8 @@ void UniqueIDBDatabase::putOrAdd(const IDBRequestData& requestData, const IDBKey
     IDBError error;
     bool usedKeyIsGenerated = false;
     uint64_t keyNumber;
-    auto transactionIdentifier = requestData.transactionIdentifier();
+    ASSERT(requestData.transactionIdentifier());
```

Comment:
We could just return callback(InvalidStateError) like the other cases above.  Same a few other places in this file.

Reply:
The reason I added assert instead of early return is because we should already validate transaction identifier at this point (when receiving the message). Let me land this first to unblock testing and I will make a followup to split IDBRequestData, which is likely a bigger refactoring.